core/trino-main/src/main/java/io/trino/server/ui/FormUiAuthenticatorModule.java

```diff
@@ -37,10 +37,10 @@ public FormUiAuthenticatorModule(boolean usePasswordManager)
     @Override
     public void configure(Binder binder)
     {
-        binder.bind(PasswordAuthenticatorManager.class).in(SINGLETON);
         binder.bind(FormWebUiAuthenticationFilter.class).in(SINGLETON);
         binder.bind(WebUiAuthenticationFilter.class).to(FormWebUiAuthenticationFilter.class).in(SINGLETON);
         if (usePasswordManager) {
+            binder.bind(PasswordAuthenticatorManager.class).in(SINGLETON);
```

Comment:
commit that makes `PasswordAuthenticatorManager` optional binding should be first.